UI/org.eclipse.birt.report.designer.ui/src/org/eclipse/birt/report/designer/internal/ui/command/SearchHandler.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2004 Actuate Corporation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *
+ * Contributors:
+ *  Actuate Corporation  - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.birt.report.designer.internal.ui.command;
+
+import org.eclipse.birt.report.designer.internal.ui.views.actions.SearchAction;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.jface.viewers.TreeViewer;
+
+/**
+ *
+ */
+
+public class SearchHandler extends SelectionHandler {
+	public static TreeViewer treeViewer = null;
+
+	/*
+	 * (non-Javadoc)
+	 *
+	 * @see
+	 * org.eclipse.core.commands.AbstractHandler#execute(org.eclipse.core.commands.
+	 * ExecutionEvent)
+	 */
+	@Override
+	public Object execute(ExecutionEvent event) throws ExecutionException {
+		super.execute(event);
+		SearchAction searchAction = new SearchAction(treeViewer);
+		searchAction.run();
+		return Boolean.TRUE;
```

Comment:
If there is not enough context then you are no doubt not using the right design.  After all, the treeViewer might be null and it might be disposed.